### PR TITLE
feat: resize panels

### LINKED ISSUE #1563: Notebooks: Resize Flux Editor Panel
We would like ability to resize the vertical height of a flux editor panel in Notebooks. Using the [LargeDraggableResizer](https://influxdata.github.io/clockface/?path=/story/layout-draggable-resizer-family--draggableresizer), a user should be able to resize the vertical height of the panel, while maintaining the original minimum height behavior.

## Acceptance Criteria

* When in a notebook with a flux editor panel, a user should see a LargeDraggableResizer and have the ability to click and drag on the resizer. After clicking and dragging, the user should be able to increase the vertical height of the panel while maintaining the original behavior of the minimum height.

### LINKED ISSUE #1564: Notebooks: Resize Metric Selector Panel
We would like ability to resize the vertical height of a metic selector panel in Notebooks. Using the [LargeDraggableResizer](https://influxdata.github.io/clockface/?path=/story/layout-draggable-resizer-family--draggableresizer), a user should be able to resize the vertical height of the panel, while maintaining the original minimum height behavior.

## Acceptance Criteria

* When in a notebook with a metric selector panel, a user should see a LargeDraggableResizer and have the ability to click and drag on the resizer. After clicking and dragging, the user should be able to increase the vertical height of the panel while maintaining the original behavior of the minimum height.

### LINKED ISSUE #1565: Notebooks: Resize Query Builder Panel
We would like ability to resize the vertical height of a query builder panel in Notebooks. Using the [LargeDraggableResizer](https://influxdata.github.io/clockface/?path=/story/layout-draggable-resizer-family--draggableresizer), a user should be able to resize the vertical height of the panel, while maintaining the original minimum height behavior.

## Acceptance Criteria

* When in a notebook with a query builder panel, a user should see a LargeDraggableResizer and have the ability to click and drag on the resizer. After clicking and dragging, the user should be able to increase the vertical height of the panel while maintaining the original behavior of the minimum height.

### LINKED ISSUE #1566: Notebooks: Resize Markdown Panel
We would like ability to resize the vertical height of a markdown panel in Notebooks. Using the [LargeDraggableResizer](https://influxdata.github.io/clockface/?path=/story/layout-draggable-resizer-family--draggableresizer), a user should be able to resize the vertical height of the panel, while maintaining the original minimum height behavior.

## Acceptance Criteria

* When in a notebook with a markdown panel, a user should see a LargeDraggableResizer and have the ability to click and drag on the resizer. After clicking and dragging, the user should be able to increase the vertical height of the panel while maintaining the original behavior of the minimum height.

### LINKED ISSUE #1567: Notebooks: Resize Visualization Panel
We would like ability to resize the vertical height of a visualization panel in Notebooks. Using the [LargeDraggableResizer](https://influxdata.github.io/clockface/?path=/story/layout-draggable-resizer-family--draggableresizer), a user should be able to resize the vertical height of the panel, while maintaining the original minimum height behavior.

## Acceptance Criteria

* When in a notebook with a visualization panel, a user should see a LargeDraggableResizer and have the ability to click and drag on the resizer. After clicking and dragging, the user should be able to increase the vertical height of the panel while maintaining the original behavior of the minimum height.

### LINKED ISSUE #1678: Notebooks: Remove the concept of a subpanel
right now we have panels that have simple table subpanels within them.

with #1676 (and #1679), we should be able to remove this weirdness, and just have... panels

this issue is to make sure we remove all of the simple table / results / resizer stuff, and the visualization query fetching, that exists in:
 - [ ] Metric Selector
 - [ ] Query Builder
 - [ ] Raw Flux Editor
 - [ ] Remote CSV (pending timing and the like)

### LINKED ISSUE #1679: Notebooks: Enable Simple Table visualization
right now simpletable is disabled, we wanna move it over to being a feature flagged visualization option... i think the flag is `simpleTable`

in doing this, we also want to make an exception in the visualization panel. i dont think it makes sense to have the aggregate window stuff in there for this visualization type, but instead we should hide that interaction when the visualization type is simple table and a `|> limit(100)` should be automatically applied to the visualization query (`append(__CURRENT_RESULT__ |> limit(100)`)

seems like this last part might come and bite me later, but aggregate window is only used to limit result sets of the panel itself, which makes me think that it should be tied to the visualization type anyways?

### LINKED ISSUE #1680: Notebooks: be able to resize all panels
with (https://github.com/influxdata/ui/issues/1678) being done, we can now treat panels universally in the system.

remove whatever is on visualizations, and add a universal resizer to the base panel class. panels all start we a default height (i think the const already exists somewhere), they can be dragged, that height is persisted on the panel data in the backend

### PR DESCRIPTION
**Pull request by drdelambre**
Closes #1567
Closes #1566
Closes #1563
Closes #1565
Closes #1564
Closes #1679
Closes #1678
Closes #1680 

twodotoh a influxcloud dev local_orgs_ca330100309e96c9_notebooks_07b04517a4953000(full up)

### CODE REVIEW

**Comment by Asalem1** [comment]
Video/gif or it didn't happen

File: src/flows/components/panel/FlowPanel.tsx [hunk 1]
```diff
@@ -108,6 +120,61 @@ const FlowPanel: FC<Props> = ({id, controls, persistentControls, children}) => {
...
+
+    updateSize(updatedHeight)
+  }
+
+  const handleMouseDown = (): void => {
+    setIsDragging(true)
+    const body = document.getElementsByTagName('body')[0]
+    body && body.classList.add('panel-resizer-dragging')
+
+    window.addEventListener('mousemove', handleMouseMove)
```
Comment (Asalem1) [inline src/flows/components/panel/FlowPanel.tsx:165]: do we want metrics on how often this is used?

File: src/flows/components/panel/Handle.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      onMouseDown={handleMouseDown}
+      title="Drag to resize"
+      data-testid={testID}
+      ref={dragRef}
+    >
+      <div className="cf-draggable-resizer--handle-pill1 cf-draggable-resizer--handle-pill1--horizontal"></div>
+      <div
+        className="cf-draggable-resizer--gradient1 cf-draggable-resizer--gradient1--horizontal"
+        style={{
+          background: `linear-gradient(to right, #ce58eb 0%, #00C9FF 100%)`,
```
Comment (Asalem1) [inline src/flows/components/panel/Handle.tsx:41]: why can't this be achieved in the `scss` file?

File: src/flows/components/panel/Results.tsx [hunk 3]
```diff
@@ -1,91 +0,0 @@
...
-        text="CSV"
-        icon={IconFont.Download}
-        onClick={download}
-        status={queryText ? ComponentStatus.Default : ComponentStatus.Disabled}
-      />
-    </Resizer>
-  )
-}
-
-export default Results
```
Comment (Asalem1) [inline src/flows/components/panel/Results.tsx:91]: RIP

File: src/visualization/types/SimpleTable/options.tsx [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import React, {FC} from 'react'
+const SimpleTableOption: FC = () => {
+  return <h5>There is nothing to configure here</h5>
+}
+
+export default SimpleTableOption
```
Comment (Asalem1) [inline src/visualization/types/SimpleTable/options.tsx:6]: question - did SimpleTable get bounced to giraffe yet?

**Comment by drdelambre** [comment]
https://www.youtube.com/watch?v=ZhZzlrEOusA

Peek 2021-06-17 09-48

File: src/flows/components/panel/FlowPanel.tsx (hunk 1 above)
Comment (drdelambre) [reply src/flows/components/panel/FlowPanel.tsx:165]: def not how muched it moved, but idk how much i care if someone dragged something around right now

File: src/flows/components/panel/Handle.tsx (hunk 2 above)
Comment (drdelambre) [reply src/flows/components/panel/Handle.tsx:41]: should be handled in clockface, but it isn't. so this is a stand in while we try to talk about exposing the drag handle in that library. wanted to have the least amount of impact so i can just delete a single file when that time comes

File: src/visualization/types/SimpleTable/options.tsx (hunk 4 above)
Comment (drdelambre) [reply src/visualization/types/SimpleTable/options.tsx:6]: not yet. working on it

File: src/flows/components/panel/Handle.tsx (hunk 2 above)
Comment (Asalem1) [reply src/flows/components/panel/Handle.tsx:41]: Sounds good. Is there a clockface issue to track this to get it out?

**Review by Asalem1 — APPROVED** [review]
tests *cough*

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=master)

**Review by genehynson — CHANGES_REQUESTED** [review]
there's a lot of code changed here...could we get a few tests before merging? <3

**Comment by jzahnd** [comment]
I'd suggest that we are ok to merge and create a follow up issue for tests given that we will be may be modifying a good portion of this behavior based on user feedback?

**Comment by genehynson** [comment]
> I'd suggest that we are ok to merge and create a follow up issue for tests given that we will be may be modifying a good portion of this behavior based on user feedback?

I guess that's fair if it's actually taking this much time to write those tests. Lets just create the issue for doing so before we merge this PR so hopefully we don't forget later on.

**Review by genehynson — APPROVED** [review]
👍

**Pull request merged**